commitlint.config.js

```diff
@@ -1 +1,54 @@
-module.exports = { extends: ["@commitlint/config-conventional"] };
+// https://github.com/conventional-changelog/commitlint/blob/master/@commitlint/config-conventional/index.js
+// https://github.com/conventional-changelog/commitlint/blob/master/docs/reference-rules.md
+module.exports = {
+  extends: ["@commitlint/config-conventional"],
+  plugins: ["commitlint-plugin-function-rules"],
+  rules: {
+    "type-enum": [
+      2,
+      "always",
+      // in addtition to angular defaults, we add additional type for commits, breaking, breaking_change will force a major release with semantic-release
+      // any body of commit containing BREAKING_CHANGE, BREAKING, etc.. will do too, please refer to the .releaserc.json file
+      [
+        "revert",
+        "feat",
+        "test",
+        "perf",
+        "fix",
+        "chore",
+        "ci",
+        "docs",
+        "breaking_change",
+        "breaking",
+        "refactor",
+      ],
+    ],
+    "subject-case": [1, "always", "lower-case"],
+    // disable it to allow release notes from semantic-release to happen when releasing
+    "footer-max-line-length": [0],
+    "body-max-line-length": [0],
+    // level: disabled to use our own rule with commitlint-plugin-function-rules plugin
+    "scope-case": [0],
+    "function-rules/scope-empty": [
+      2,
+      "always",
+      (parsed) => {
+        const ticketNumberFormat = /^cb2-\d+$/;
```

Comment:
This will need to be changed to account for the VTA branch names for example: VTA-12312 will produce a validation error.

Reply:
Very good point. I can see this being an issue for different teams working in the same repositories. I'll discuss this with the team.

Reply:
Removed hard coding of cb2.